### PR TITLE
Get comments method fix

### PR DESCRIPTION
**Pull request by gecevicius**
The GetComments method occasionally returned a 500 error. This was because there was a self referencing loop in the Comments model. Fixed by creating a new view model for the GetComments method.

### CODE REVIEW

File: Inspicio/Controllers/ImagesController.cs [hunk 1]
```diff
@@ -288,15 +288,25 @@ public JsonResult GetRating(int? id)
         }
         public JsonResult GetComments(int? Id)
         {
-            List<CommentInfo> comments = new List<CommentInfo>();
-            var AllComments = _context.Comments.Where(c => c.ScreenId == Id);
+            List<CommentsViewModel> comments = new List<CommentsViewModel>();
+            var AllComments = _context.Comments.Where(c => c.ScreenId == Id).Include(x => x.ApplicationUser).ThenInclude(y => y.Comments);
```
Comment (wilka) [inline Inspicio/Controllers/ImagesController.cs]: The `Include(x => x.ApplicationUser).ThenInclude(y => y.Comments)` part probably isn't needed, it looks like it's only `ApplicationUser` that's used.

File: Inspicio/Controllers/ImagesController.cs [hunk 2]
```diff
@@ -288,15 +288,25 @@ public JsonResult GetRating(int? id)
...
+                CommentsViewModel.ScreenId = SingleComment.ScreenId;
+                CommentsViewModel.Lat = SingleComment.Lat;
+                CommentsViewModel.Lng = SingleComment.Lng;
+                CommentsViewModel.Message = SingleComment.Message;
+                CommentsViewModel.Timestamp = SingleComment.Timestamp;
+                CommentsViewModel.ParentId = SingleComment.ParentId;
+                comments.Add(CommentsViewModel);
             }
+
+            var jsonText = Newtonsoft.Json.JsonConvert.SerializeObject(comments);
```
Comment (wilka) [inline Inspicio/Controllers/ImagesController.cs]: This `JsonConvert.SerializeObject` looks like it shouldn't be here.

File: Inspicio/Models/Comment.cs [hunk 3]
```diff
@@ -4,6 +4,7 @@
 using System.ComponentModel.DataAnnotations.Schema;
 using System.Linq;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
```
Comment (wilka) [inline Inspicio/Models/Comment.cs]: This `using` doesn't seem to be needed

**Review by wilka — CHANGES_REQUESTED** [review]

**Comment by wilka** [comment]
This looks like there's a few different things going on in this pull request, it's good to keep feature branches only doing one (if it's possible)

File: Inspicio/Models/Comment.cs (hunk 3 above)
Comment (gecevicius) [reply Inspicio/Models/Comment.cs]: Forgot to delete this after trying out the [JsonIgnore] attribute.

**Comment by wilka** [comment]
Adding comments via pins doesn't seem to work, I'm not sure if that's new in this pull request?

The comments are added to the sidebar, but the pin isn't displayed over the image after the comment is added. I tried clickling the show pins button, but that didn't help. Clicking "open pin" on the comment didn't work either.

pin

**Comment by wilka** [comment]
👍 It looks like e420821e15b010ec37e9d0bcc16ad8691a00e049 has fixed that pin bug.

**Review by wilka — APPROVED** [review]

**Pull request merged**